src/Api/Endpoints/Register.cs

```diff
@@ -31,8 +32,15 @@ public static void MapRegisterEndpoints(this WebApplication app)
             .RequireCors("default")
             .WithMetadata(new HttpMethodMetadata(new string[] { "POST" }, acceptCorsPreflight: true));
 
-        app.MapPost("/register/complete", async (RegistrationCompleteDTO payload, HttpRequest request, IFido2ServiceFactory fido2ServiceFactory) =>
+        app.MapPost("/register/complete", async (RegistrationCompleteDTO payload, HttpRequest request, IFido2ServiceFactory fido2ServiceFactory, IValidator<RegistrationCompleteDTO> validator) =>
         {
+            var validation = await validator.ValidateAsync(payload);
+
+            if (!validation.IsValid)
+            {
+                throw new ValidationException(validation.Errors);

```

Comment:
I'd rather just return `return ValidationProblems` from here instead of throwing.

Reply:
Made the change.  I didn't know what to do with the `detail` or `instance` parameters. So I left them `null` for now since that's the default.